framework/src/fviks/FVScalarLagrangeMultiplierInterface.C

```diff
@@ -0,0 +1,111 @@
+  mooseAssert(((ft1 == ft_both) && (ft2 == ft_both)) ||
+                  (_elem_is_one && (ft1 == ft_elem) && (ft2 == ft_neigh)) ||
+                  (!_elem_is_one && (ft1 == ft_neigh) && (ft2 == ft_elem)),
+              "These seem like the reasonable combinations of face types.");
+#endif
+  mooseAssert(_lambda.size() == 1 && _lambda_var.order() == 1,
+              "The lambda variable should be first order");
+  mooseAssert(_lambda_var.dofIndices().size() == 1, "We should only have a single dof");
+
+  const auto var_elem_num = _elem_is_one ? var1().number() : var2().number();
+  const auto var_neigh_num = _elem_is_one ? var2().number() : var1().number();
+
+  const auto r = MetaPhysicL::raw_value(_lambda[0]) * fi.faceArea() * fi.faceCoord();
+
+  // Primal residual
+  prepareVectorTag(_assembly, var_elem_num);
+  mooseAssert(_local_re.size() == 1, "We should only have a single dof");
+  _local_re(0) = r;
+  accumulateTaggedLocalResidual();
+  prepareVectorTagNeighbor(_assembly, var_neigh_num);
+  _local_re(0) = -r;
+  accumulateTaggedLocalResidual();
+
+  // LM residual. We may not have any actual ScalarKernels in our simulation so we need to manually
+  // make sure the scalar residuals get cached for later addition
+  const auto lm_r = MetaPhysicL::raw_value(computeQpResidual()) * fi.faceArea() * fi.faceCoord();
+  _assembly.cacheResidual(_lambda_var.dofIndices()[0], lm_r, _vector_tags);
+}
+
+void
+FVScalarLagrangeMultiplierInterface::computeJacobian(const FaceInfo & fi)
+{
+  _face_info = &fi;
+  _normal = fi.normal();
+  _elem_is_one = sub1().find(fi.elem().subdomain_id()) != sub1().end();
+
+#ifndef NDEBUG
+  const auto ft1 = fi.faceType(var1().name());
+  const auto ft2 = fi.faceType(var2().name());
+  constexpr auto ft_both = FaceInfo::VarFaceNeighbors::BOTH;
+  constexpr auto ft_elem = FaceInfo::VarFaceNeighbors::ELEM;
+  constexpr auto ft_neigh = FaceInfo::VarFaceNeighbors::NEIGHBOR;
+  mooseAssert(((ft1 == ft_both) && (ft2 == ft_both)) ||
+                  (_elem_is_one && (ft1 == ft_elem) && (ft2 == ft_neigh)) ||
+                  (!_elem_is_one && (ft1 == ft_neigh) && (ft2 == ft_elem)),
+              "These seem like the reasonable combinations of face types.");
+#endif
+  mooseAssert(_lambda.size() == 1 && _lambda_var.order() == 1,
+              "The lambda variable should be first order");
+
+  const auto & elem_dof_indices = _elem_is_one ? var1().dofIndices() : var2().dofIndices();
+  const auto & neigh_dof_indices =
+      _elem_is_one ? var2().dofIndicesNeighbor() : var1().dofIndicesNeighbor();
+  mooseAssert((elem_dof_indices.size() == 1) && (neigh_dof_indices.size() == 1),
+              "We're currently built to use CONSTANT MONOMIALS");
+
+  // Primal
+  const auto primal_r = _lambda[0] * fi.faceArea() * fi.faceCoord();
+  _assembly.processDerivatives(primal_r, elem_dof_indices[0], _matrix_tags);
+  _assembly.processDerivatives(-primal_r, neigh_dof_indices[0], _matrix_tags);
```

Comment:
This is the equation that will determine the sign of the LM variable. Which variable is the element variable will determine what the sign of `primal_r` needs to be and consequently what the lambda sign needs to be. You can imagine that if you swap which variable is element, the flux from 1 to 2 does not change, but the sign of `primal_r` and the LM must change. If you wanted to make this independent of processor count, then you'd have to multiply this `primal_r` by say 1 if `_elem_is_one` and -1 if not.

Reply:
I do agree that it is the primal variable values that matter, so I am fine with the solution of hiding the lambda output and not changing anything here.

Reply:
This is the same as in FVInterfaceKernel though?
We look at which variable lives on elem 1, and add +residual to it.

I guess I had misunderstood it initially.

Reply:
In all our other flux objects a positive flux (e.g. a positive return quantity from `computeQpResidual()`) denotes a flux moving in the direction from `elem` to `neighbor` because that is how our normals are defined. Normals *must* be defined in a way that is irrespective of variables because it's a geometric feature. I think we want to continue to define a flux that is positive to be a flux that is moving in the normal direction.

Reply:
Ok. Kind of having trouble thinking in terms of fluxes for the lagrange multiplier.
Either way, I will fix this, because it ll pop back up at the person making a penalty interface.